scripts/verify.py

```
@@ -576,10 +595,21 @@ def run_step(step: Step) -> dict[str, object]:
+    # Diagnostic instrumentation (round 20, Part B) -- capture_output=True
+    # instead of the previous bare subprocess.run(...) (which relied on the
+    # child inheriting this process's stdout/stderr fds directly). Explicitly
+    # capturing and then unconditionally re-printing the child's own
+    # stdout/stderr here, regardless of exit status, means a failing step's
+    # diagnostic output can never be silently lost to an intermediate
+    # buffering/redirection layer between the child and whatever ultimately
+    # renders this script's output (e.g. a CI runner's non-tty log pipe) --
+    # see this module's own top-of-file comment for the full reasoning. This
+    # also means a caller with `--json` gets the raw text available even when
+    # the terminal itself scrolled it out of view.
+    proc = subprocess.run(step.cmd, cwd=ROOT, env=env, capture_output=True, text=True)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)